addons/sourcemod/scripting/szf/pickupweapons.sp

```diff
@@ -104,7 +106,44 @@ public Action Event_WeaponsRoundStart(Event event, const char[] name, bool dontB
 			}
 			case WeaponType_RareSpawn:
 			{
-				SetRandomWeapon(iEntity, WeaponRarity_Rare);
+				if (aWeaponsRares.Length > 0)
+				{
+					//Make sure every spawn weapon is different
+					int iRandom = GetRandomInt(0, aWeaponsRares.Length - 1);
+					
+					Weapon wep;
+					aWeaponsRares.GetArray(iRandom, wep);
+					
+					SetWeaponModel(iEntity, wep);
+					aWeaponsRares.Erase(iRandom);
+				}
+				else
+				{
+					//If we already went through every spawn weapons, no point having rest of it
+					RemoveEntity(iEntity);
+					continue;
+				}
```

Comment:
Create a new function for this instead of duplicating the code. You can pass the ArrayList, or clone it.